credhub-format.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+set -eo pipefail
+
+# yq operates on yaml documents using a jq-like syntax.
+# For jq program syntax, see: https://stedolan.github.io/jq/manual/
+
+if [ -z $2 ]; then
+  echo "
+Usage: ./credhub-format.sh <source-file> <pipeline-name>
+Example: ./credhub-format.sh credentials.yml deploy-something > output.yml
+
+Read from a credential file and reformat the contents to a JSON format that CredHub
+can import, excluding any keys that do not appear in ci/pipeline.yml.
```

Comment:
So is the main point of the script that it excludes credentials which aren't used in the pipeline?

Reply:
The main point is to convert from our existing yaml credential files format:

```yaml
key: value
```

to the format Credhub's `import` command expects, JSON that looks like: 

```json
{
  credentials: [
    {
      "name": /concourse/main/repo-name/key,
      "type": "value",
      "value": <value>,
    },
    ...
  ]
}
```

The exclusion of unused keys is a bonus to make sure we're not carrying cruft into the new system.